Always provide the vector metatable to node callbacks

j-r opened the pull request:
Use vector.new/copy/offset as appropriate and replace deprecated use of vector.new by vector.copy.

### Background

While doing their redstone rewrite Mineclonia decided to make aggressive use of the vector metatable and assume that it is present when positions are passed to standard callbacks (see https://codeberg.org/mineclonia/mineclonia/issues/3820). This worked out surprisingly well and it seems feasible to update mods that don't always provide the metatable. Digtron already uses vector.new in some places and this PR consistently replaces all manually created position vectors with the appropriate vector constructor.

### Testing

There should be no change in digtron behavior.

j-r committed:
Always provide the vector metatable to node callbacks

Use vector.new/copy/offset as appropriate and replace deprecated use of
vector.new by vector.copy.

SmallJoker commented on class_layout.lua:
```diff
@@ -5,7 +5,7 @@ digtron.DigtronLayout.__index = digtron.DigtronLayout
 
 local get_node_image = function(pos, node)
-	local node_image = {node=node, pos={x=pos.x, y=pos.y, z=pos.z}}
+	local node_image = {node=node, pos=vector.copy(pos)}
```
`vector.copy` requires Minetest 5.5.0: https://github.com/luanti-org/luanti/commit/2cefe51d3b9bc4f3ae18854e171a06ea83e9cb25

Please add `min_minetest_version` to `mod.conf` such that it is picked up by ContentDB automatically. Furthermore, a presence check with a pleasant `error("Your Luanti/Minetest version is too old. Please update to 5.5.0 or newer.")` user "message" upon startup would be nice to have such that there's no issues/errors during runtime.

j-r committed:
Add min_minetest_version = 5.5 and error out on older versions

But provide fallback to run with default mods that don't have
set_inventory_action_loggers.

j-r commented:
Added a commit that adds a message when run on old Luanti or MTG versions as well as adding polyfills, and a 2nd one with `min_minetest_version`.

I was quite happy with these until I noticed that fakelib is already using `vector.copy`. Aargh:-)

I think I'll mostly remove the polyfill commit and just keep the message (upgraded to an error) after testing for `vector.copy` and `default.set_inventory_action_loggers` and merge with the min version commit. This should also fix the luacheck problem.

SmallJoker reviewed (approved):
Looks good. Thank you for the adjustments.
Will merge in a few days unless there are objections.
